### PR TITLE
21481 remove group clickable area

### LINKED ISSUE #21481: "Remove group" button is not fully clickable
**Describe the bug**
When deleting a permissions group, the `Remove group` button only works if you click the text, but not if you click any other part of the button.

**To Reproduce**
1. Go to Admin -> People -> Groups
2. Create a new group
3. Click on `...` next to the new group and click on the "Remove Group" button, in the space to the right of the text

**Expected behavior**
You should be able to click anywhere in the blue area on the button. Strangely, the "Edit Name" button above it works correctly here.

**Screenshots**
https://user-images.githubusercontent.com/32746338/161888242-b5934ac0-a0cc-40d0-b24d-f707829499f1.mov

**Severity**
Mild but confusing, since if you click on the `...` then move your cursor directly down to "Remove Group" the button won't work.

### PR DESCRIPTION
**Pull request by losrebellos**
Fixes #21481.

__After__

With the whole `Remove group` area clickable.

image

### CODE REVIEW

File: frontend/src/metabase/components/Triggerable.jsx [hunk 1]
```diff
@@ -133,7 +138,8 @@ export default ComposedComponent =>
       }
 
       return (
-        <a
+        <Trigger
+          as={as}
```
Comment (losrebellos) [inline frontend/src/metabase/components/Triggerable.jsx:142]: Add `as` property so the Trigger's tag can customised

File: frontend/src/metabase/components/Triggerable.jsx [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { isObscured } from "metabase/lib/dom";
 import Tooltip from "./Tooltip";
 
 import cx from "classnames";
+import styled from "@emotion/styled";
+
+const Trigger = styled.a``;
```
Comment (losrebellos) [inline frontend/src/metabase/components/Triggerable.jsx:11]: `Trigger` using emotion so we can use the `as` property

File: frontend/src/metabase/admin/people/components/GroupsListing.jsx [hunk 3]
```diff
@@ -87,16 +87,26 @@ function ActionsPopover({ group, onEditGroupClicked, onDeleteGroupClicked }) {
...
         >
           {t`Edit Name`}
         </li>
-        <li className="pt1 pb2 px2 bg-brand-hover text-white-hover cursor-pointer text-error">
-          <ModalWithTrigger triggerElement={t`Remove Group`}>
-            <DeleteGroupModal group={group} onConfirm={onDeleteGroupClicked} />
-          </ModalWithTrigger>
-        </li>
+        <ModalWithTrigger
+          as="li"
```
Comment (losrebellos) [inline frontend/src/metabase/admin/people/components/GroupsListing.jsx:96]: ModalWithTrigger rendered as a `li` tag

File: frontend/src/metabase/admin/people/components/GroupsListing.jsx [hunk 4]
```diff
@@ -87,16 +87,26 @@ function ActionsPopover({ group, onEditGroupClicked, onDeleteGroupClicked }) {
     >
       <ul className="UserActionsSelect">
         <li
-          className="pt1 pb2 px2 bg-brand-hover text-white-hover cursor-pointer"
+          className="pt2 pb2 px2 bg-brand-hover text-white-hover cursor-pointer"
```
Comment (losrebellos) [inline frontend/src/metabase/admin/people/components/GroupsListing.jsx:90]: Padding fix for triggers

File: frontend/src/metabase/admin/people/components/GroupsListing.jsx [hunk 5]
```diff
@@ -87,16 +87,18 @@ function ActionsPopover({ group, onEditGroupClicked, onDeleteGroupClicked }) {
     >
       <ul className="UserActionsSelect">
         <li
-          className="pt1 pb2 px2 bg-brand-hover text-white-hover cursor-pointer"
+          className="py1 px2 bg-brand-hover text-white-hover cursor-pointer"
```
Comment (losrebellos) [inline frontend/src/metabase/admin/people/components/GroupsListing.jsx:90]: Align padding with other `UserActionsSelect`

File: frontend/src/metabase/css/admin.css [hunk 6]
```diff
@@ -264,7 +264,6 @@
 }
 
 .UserActionsSelect {
-  padding-top: 1em;
```
Comment (losrebellos) [inline frontend/src/metabase/css/admin.css:267]: Remove `padding-top` as it creates a "strange" white space at the top of the popover menu

File: frontend/src/metabase/components/Triggerable.jsx (hunk 2 above)
Comment (alxnddr) [reply frontend/src/metabase/components/Triggerable.jsx:11]: Nice 👏

**Review by alxnddr — APPROVED** [review]

**Pull request merged**